### PR TITLE
Feature/api v2 drf

### PR DESCRIPTION
**Pull request by brianjgeiger**
Make the first publicly accessible API for the OSF.

### CODE REVIEW

File: api/base/authentication/drf.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    """Given a cookie value, return the `Session` object or `None`."""
+    session_id = itsdangerous.Signer(settings.SECRET_KEY).unsign(cookie_val)
+    return Session.load(session_id)
+
+# http://www.django-rest-framework.org/api-guide/authentication/#custom-authentication
+class OSFSessionAuthentication(authentication.BaseAuthentication):
+    """Custom DRF authentication class which works with the OSF's Session object.
+    """
+
+    def authenticate(self, request):
```
Comment (icereval) [inline api/base/authentication/drf.py]: We'll need to switch this over to CAS authentication, lets review in the morning if time allows, currently this depends on the cookie from osf (obviously).

File: api/base/authentication/drf.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        if not session:
+            return None
+        user_id = session.data.get('auth_user_id')
+        user = User.load(user_id)
+        if user:
+            return user, None
+        return None
+
+
+class OSFBasicAuthentication(BasicAuthentication):
```
Comment (icereval) [inline api/base/authentication/drf.py]: I loaded this up for testing and don't see where we send the username/password via a login form anywhere? Also this does not check for one time password, replacing with CAS auth will fix this.

Comment (brianjgeiger) [inline api/base/authentication/drf.py]: Basic auth is solely for unit testing.

Comment (sloria) [inline api/base/authentication/drf.py]: @brianjgeiger Speaking of which, we should have tests for each of our allowed authn policies. Is this in the works?

Comment (brianjgeiger) [inline api/base/authentication/drf.py]: Basic is pretty well tested. ;) nothing yet, but it can be added.

Comment (sloria) [inline api/base/authentication/drf.py]: Cool. webtest-plus has built-in support for Basic Auth and Bearer Tokens. Let me know if anything else is needed (or send a PR to webtest-plus =) ).

**Pull request merged**